signac/core/synced_list.py

```diff
@@ -136,6 +139,7 @@ def reset(self, data=None):
 
     def __setitem__(self, key, value):
         self.load()
+        value = self._validate(value)
```

Comment:
Should the item be validated before loading and suspending synchronization? Seems like it should be the first step. Also for all the other methods below.